app/scripts/migrations/115.ts

```diff
@@ -0,0 +1,66 @@
+import { cloneDeep, isEmpty } from 'lodash';
+import {
+  TransactionMeta,
+  TransactionStatus,
+} from '@metamask/transaction-controller';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 115;
+
+// Target date is December 8, 2023 - 00:00:00 UTC
+export const TARGET_DATE = new Date('2023-12-08T00:00:00Z').getTime();
+
+/**
+ * This migration updates the status of transactions that were approved or signed before December 8, 2023 to failed.
```

Comment:
I think your PR description says it better. Something like:
```suggestion
 * This migration sets the `status` to `failed` for all transactions created before December 8, 2023 that are still `approved` or `signed`.
```